htlcswitch/circuit.go

```diff
@@ -114,16 +59,22 @@ func (m *circuitMap) add(circuit *PaymentCircuit) error {
 		}
 
 	} else {
-		m.circuits[circuit.PaymentHash] = make([]*PaymentCircuit, 1)
+		m.circuits[circuit.PaymentHash] = make([]*channeldb.PaymentCircuit, 1)
+	}
+
+	// If database isn't empty we will sync circuit map with database, by
+	// saving it.
+	if err := m.db.AddCircuit(circuit); err != nil {
```

Comment:
The comment here doesn't match up with the code implemented. To fix this, the `db` pointer should be checked for nil-ness.